core/2d/ProgressTimer.cpp

```diff
@@ -231,8 +231,21 @@ void ProgressTimer::updateDisplayedOpacity(uint8_t parentOpacity)
 {
     Node::updateDisplayedOpacity(parentOpacity);
 
-    _sprite->updateDisplayedOpacity(_displayedOpacity);
+    _displayedOpacity = _realOpacity * parentOpacity / 0xFF;
+    
+    _sprite->setOpacity(_displayedOpacity);
+    updateColor();
     updateProgress();
+    
+    if (_cascadeOpacityEnabled)
+    {
+        _sprite->updateDisplayedOpacity(_displayedOpacity);
```

Comment:
I may be wrong, but since `_sprite` is considered part of `ProgressTimer` widget, then shouldn't `_sprite->updateDisplayedOpacity(_displayedOpacity);` be always called? As I understand `_cascadeOpacityEnabled` part should only apply for the children.